Keas.Mvc/Services/IdentityService.cs

```diff
@@ -80,5 +86,83 @@ public async Task<User> GetByKerberos(string kerb)
+                    {
+                        user = await _context.Users.SingleOrDefaultAsync(u => u.Id == kerbResults.ResponseData.Results[0].UserId);
+                        if (user == null)
+                        {
+                            // User not found with Kerb Id either. Add user
+                            var newUser = new User()
+                            {
+                                FirstName = kerbResults.ResponseData.Results[0].DFirstName,
+                                LastName = kerbResults.ResponseData.Results[0].DLastName,
+                                Id = kerbResults.ResponseData.Results[0].UserId,
+                                Email = contactResult.ResponseData.Results[0].Email,
+                                Iam = id.IamId
+                            };
+                            if (newUser.Id == null || newUser.Email == null || newUser.FirstName == null || newUser.LastName == null)
+                            {
+                                warning.Append("User could not be added: IAM ID: " + id.IamId.ToString() + " Name: " + newUser.FirstName + " " + newUser.LastName + " | ");
+                            }
+                            else
+                            {
+                                _context.Users.Add(newUser);
+                                await SavePerson(team, newUser);
+                                newpeople += 1;
+                            }
+                        }
+                        else
+                        {
+                            // User existed with Kerb Id, check team
+                            if (!await _context.People.AnyAsync(p => p.UserId == user.Id && p.Team.Slug == teamslug))
+                            {
+                                await SavePerson(team, user);                                
+                                newpeople += 1;
+                            }
+                        }
+                    }
+                }
+                else if (!await _context.People.AnyAsync(p => p.User.Iam == id.IamId && p.Team.Slug == teamslug))
+                {
+                    // User exists with IAM ID, but not in this team
+                    await SavePerson(team, user);
+                    newpeople += 1;
+                }
+            }
+            string returnMessage = newpeople.ToString() + " new people added to the team. " + warning.ToString();
+            return returnMessage;
+        }
+
+        private async Task SavePerson(Team team, User user)
+        {
+            var newPerson = new Person()
+            {
+                User = user,
+                Team = team,
+                FirstName = user.FirstName,
+                LastName = user.FirstName,
+                Email = user.Email
+            };
+            _context.People.Add(newPerson);
+            await _context.SaveChangesAsync();
```

Comment:
why save changes a bunch of times instead of all at once after people are added to the context?  does IAM return the same person multiple times or some other issue?